server/routes/api/graphql.ts

```diff
@@ -45,34 +46,43 @@ const resolvers: IResolver = {
 				})
 				.limit(args.n);
 
-			return allUsers.map(userRecordToGraphql);
+			return await Promise.all(allUsers.map(userRecordToGraphql));
 		},
 		search_user: async (prev, args) => {
 			let escapedQuery: string = args.search;
 			if (!args.use_regex) {
 				escapedQuery = escapedQuery.trim().replace(/[|\\{()[^$+*?.-]/g, "\\$&");
 			}
 			const queryRegExp = new RegExp(escapedQuery, "i");
-
+			const query = [
+				{
+					name: {
+						$regex: queryRegExp
+					}
+				},
+				{
+					email: {
+						$regex: queryRegExp
```

Comment:
Do we have to worry about escaping here since this is user supplied input?

Reply:
Escaping happens on line 54 right above this. Input is normally escaped unless the user ticks the Use Regex box in which case it is directly passed through. I don't see any security implications to doing that because the regular expressions only retrieve data and because this feature is already restricted to admins only.